staff/lab/ocf-tv

```diff
@@ -43,15 +43,25 @@ def wait_for_port(host, port, timeout=5, ssh_proc=None):
 
 
 def volume(args):
-    call(('ssh', args.host,
-          'export PULSE_SERVER=localhost; pactl set-sink-volume {} {}%'.format(args.sink, args.amount)))
+    cmd = 'export PULSE_SERVER=localhost; '
+
+    if args.amount is not None:
+        cmd += 'pactl set-sink-volume 0 {}%; '.format(args.amount)
+
+    # awk prints the 9th line after the match
+    cmd += "pactl list sinks | awk '/Sink #0/{nr[NR+9]}; NR in nr' | cut -d' ' -f 6;"
+
+    return call(('ssh', args.host, cmd))
 
 
 def volume_in_range(val):
-    x = int(val)
+    # allow things like +20, -30, and absolute numbers
+    x = int(val[1:]) if val[0] in ['+', '-'] else int(val)
     if x < 0 or x > 150:
```

Comment:
No, really, you don't have to do string matching. Try this (acknowledging @kpengboy's remark):
```
x = int(val)
if abs(x) < 150:
 raise ValueError('Volume out of bounds.')
return '-- ' + str(x)
```
Also, this function name is totally misleading.